web-local/src/routes/(application)/dashboard/[id]/+page.ts

```diff
@@ -26,21 +22,14 @@ export async function load({ params }) {
       return redirect(307, `/dashboard/${params.id}/edit`);
     }
   } catch (err) {
-    const invalidDashboardErrors = [
-      ExplorerSourceModelDoesntExist,
-      ExplorerSourceModelIsInvalid,
-      ExplorerSourceColumnDoesntExist,
-    ];
-
-    // if dashboard is invalid, redirect to the metrics definition page
     if (
-      invalidDashboardErrors.some(
-        (errMsg) => errMsg.includes(err.message) || err.message.includes(errMsg)
-      )
+      ExplorerDashboardDoesntExist.includes(err.message) ||
```

Comment:
Update the backend to correctly return `ExplorerMetricsDefinitionDoesntExist` and handle that here.